utils/index.d.ts

```diff
@@ -1678,3 +1682,99 @@ export declare enum AzExtResourceType {
     VirtualNetworks = 'VirtualNetworks',
     WebHostingEnvironments = 'WebHostingEnvironments',
 }
+
+/**
+ * Interface describing an object that wraps another object.
+ *
+ * The host extension will wrap all tree nodes provided by the client
+ * extensions. When commands are executed, the wrapper objects are
+ * sent directly to the client extension, which will need to unwrap
+ * them. The `registerCommandWithTreeNodeUnwrapping` method below, used
+ * in place of `registerCommand`, will intelligently do this
+ * unwrapping automatically (i.e., will not unwrap if the arguments
+ * aren't wrappers)
+ */
+export declare interface Wrapper {
+    unwrap<T>(): T;
+}
+
+/**
+ * Tests to see if something is a wrapper, by ensuring it is an object
+ * and has an "unwrap" function
+ * @param maybeWrapper An object to test if it is a wrapper
+ * @returns True if a wrapper, false otherwise
+ */
+export declare function isWrapper(maybeWrapper: unknown): maybeWrapper is Wrapper;
+
+export declare function appResourceExperience<TPick extends ContextValueFilterableTreeNode>(context: IActionContext, tdp: TreeDataProvider<ResourceGroupsItem>, resourceTypes?: AzExtResourceType | AzExtResourceType[], childItemFilter?: ContextValueFilter): Promise<TPick>;
+export declare function contextValueExperience<TPick extends ContextValueFilterableTreeNode>(context: IActionContext, tdp: TreeDataProvider<TPick>, contextValueFilter: ContextValueFilter): Promise<TPick>;
+export declare function findByIdExperience<TPick extends FindableByIdTreeNode>(context: IActionContext, tdp: TreeDataProvider<TPick>, id: string | Uri): Promise<TPick>;
+
+interface CompatibilityPickResourceExperienceOptions {
+    resourceTypes?: AzExtResourceType | AzExtResourceType[];
+    childItemFilter?: ContextValueFilter
+}
+
+export declare function compatibilityPickAppResourceExperience<TPick extends AzExtTreeItem>(context: IActionContext, tdp: TreeDataProvider<ResourceGroupsItem>, options: CompatibilityPickResourceExperienceOptions): Promise<TPick>;
+
+export declare interface QuickPickWizardContext<TNode extends unknown> extends IActionContext {
+    pickedNodes: TNode[];
+}
+
+/**
+ * Describes filtering based on context value. Items that pass the filter will
+ * match at least one of the `include` filters, but none of the `exclude` filters.
+ */
+export declare interface ContextValueFilter {
+    /**
+     * This filter will include items that match *any* of the values in the array.
+     * When a string is used, exact value comparison is done.
+     */
+    include: string | RegExp | (string | RegExp)[];
```

Comment:
I think @bwateratmsft's comment kind of got buried, but supporting just RegExp/RegExp[] feels easier to manage for us. String comparisons feel less useful now with how context values are appended to the array now. 

For backwards compat, we can just change any string context values to RegExps.

Reply:
We can get that change in after #1239.